Inventory items

davcamer opened the pull request:
Adding inventory items, using generated code.

I need to do some functional testing with this, but I believe the code is complete at the moment.

mdlayher commented on netbox/tenancy_tenants_test.go:
```diff
@@ -31,14 +31,12 @@ func TestTenantGet(t *testing.T) {
 		wantGroup *TenantGroup
```
While you're at it, can you remove this struct field since it's now unused?

davcamer commented:
I removed that struct and added myself to the AUTHORS file.

I also did some light hands-on testing. Here's a little go program that exercises the new code, listing any nodes named "node 1" (strict-matching) in netbox:
```golang
package main

import (
  "./netbox"
  "fmt"
)

func main() {
  c, _ := netbox.NewClient("http://localhost:8000", nil)
  page := c.DCIM.InventoryItems.List( &netbox.ListInventoryItemOptions{ Name: "node 1", } )

  for page.Next() {
    items, _ := c.DCIM.InventoryItems.Extract(page)
    for n, i := range items {
      fmt.Printf("%d: %+v\n", n, i)
    }
  }
}
```

mdlayher commented:
LGTM, thanks!
